app/src/main/java/com/ddoczi/tasky/authentication/data/AuthenticationRepositoryImpl.kt

```diff
@@ -0,0 +1,27 @@
+package com.ddoczi.tasky.authentication.data
+
+import com.ddoczi.tasky.authentication.data.remote.RegistrationBodyDTO
+import com.ddoczi.tasky.core.data.remote.AuthenticationApi
+import com.ddoczi.tasky.core.domain.reposotory.AuthenticationRepository
+
+class AuthenticationRepositoryImpl(
+    private val authenticationApi: AuthenticationApi
+) : AuthenticationRepository {
+    override suspend fun register(
+        fullName: String,
+        email: String,
+        password: String
+    ): Result<Unit> {
+        return try {
+            val body = RegistrationBodyDTO(
+                fullName = fullName,
+                email = email,
+                password = password
+            )
+            authenticationApi.register(body)
+            Result.success(Unit)
+        } catch (e: Exception) {
+            Result.failure(e)
+        }
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `FUNCTION_NAME`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video I show the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc